code/modules/spells/targeted/buttbots_revenge.dm

```diff
@@ -10,26 +10,41 @@
 	invocation_type = SpI_SHOUT
 	range = 1
 	cooldown_min = 200 //100 deciseconds reduction per rank
-
+	level_max = list(Sp_TOTAL = 5, Sp_SPEED = 4, Sp_POWER = 1)
 	sparks_spread = 1
 	sparks_amt = 4
 
 	amt_weakened = 8
 	amt_stunned = 8
 
 	hud_state = "wiz_butt"
+	
+	var/summon_bot = 0
 
 /spell/targeted/buttbots_revenge/cast(var/list/targets)
 	..()
 	for(var/mob/living/target in targets)
 		if(ishuman(target) || ismonkey(target))
 			var/mob/living/carbon/C = target
 			if(C.op_stage.butt != 4) // does the target have an ass
-				var/obj/item/clothing/head/butt/B = new(C.loc)
-				B.transfer_buttdentity(C)
+				if(summon_bot)
+					new /obj/machinery/bot/buttbot(C.loc)
+				else	
+					var/obj/item/clothing/head/butt/B = new(C.loc)
+					B.transfer_buttdentity(C)
 				C.op_stage.butt = 4 //No having two butts.
 				to_chat(C, "<span class='warning'>Your ass just blew up!</span>")
 			playsound(get_turf(src), 'sound/effects/superfart.ogg', 50, 1)
 			C.apply_damage(40, BRUTE, LIMB_GROIN)
 			C.apply_damage(10, BURN, LIMB_GROIN)
-	return
\ No newline at end of file
+	return
+	
+/spell/targeted/buttbots_revenge/empower_spell()
```

Comment:
I'm pretty sure it's fine, can_improve is called by the spell book. If the need for sanity arises, that check can be added to apply_upgrade() process instead (which handles all upgrades and calls procs like empower_spell)